python/ray/data/dataset.py

```diff
@@ -2294,6 +2298,12 @@ def train_test_split(
 
         if not isinstance(test_size, (int, float)):
             raise TypeError(f"`test_size` must be int or float got {type(test_size)}.")
+        
+        # Handle stratified splitting
+        if stratify is not None:
+            return self._stratified_split(ds, test_size, stratify)
```

Comment:
I think we should  disallow specifying both shuffle and stratify at the same time.